crates/bifrost/src/providers/local_loglet/mod.rs

```diff
@@ -230,14 +234,31 @@ impl LogletBase for LocalLoglet {
             Err(ShutdownError.into())
         })?;
 
-        self.last_committed_offset
-            .fetch_max(offset.into(), Ordering::Relaxed);
-        self.notify_readers();
+        // The order here is critical. We must update the last_committed_offset before checking the
+        // seal to ensure that `find_tail` observe `last_committed_offset` changes.
+        let release_pointer = LogletOffset::from(
+            self.last_committed_offset
+                .fetch_max(offset.into(), Ordering::SeqCst)
+                .max(offset.into()),
+        );
+        self.notify_readers(release_pointer);
+        // Ensure that we don't acknowledge the append (even that it has happened) if the loglet
+        // has been sealed already. Memory ordering Acquire is required to establish happens-before
+        // relationship between bumping the last_commit_offset and the seal operation.
+        if self.sealed.load(Ordering::Acquire) {
```

Comment:
I think I don't fully understand the reasoning here. If we read `sealed == true`, it makes sense to abort the append. Couldn't this read be done with `Ordering::Relaxed` since we don't depend on any other memory writes that happened before setting `sealed = true`? If we read `sealed == false`, this is probably the dangerous situation since we don't know whether the write (self.sealed = true) is still in the store buffer and we have already read `self.last_committed_offset` in `find_tail` before we incremented it in this method?